unblob/handlers/archive/hp/ipkg.py

```diff
@@ -0,0 +1,106 @@
+import io
+from pathlib import Path
+from typing import Optional
+
+from dissect.cstruct import Instance
+from structlog import get_logger
+
+from unblob.extractor import carve_chunk_to_file, is_safe_path
+from unblob.file_utils import Endian, File, InvalidInputFormat, StructParser, snull
+from unblob.models import Chunk, Extractor, HexString, StructHandler, ValidChunk
+
+logger = get_logger()
+
+C_DEFINITIONS = r"""
+    typedef struct ipkg_file_entry {
+        char name[256];
+        uint64 offset;
+        uint64 size;
+        uint32 crc32;
+    } ipkg_toc_entry_t;
+
+    typedef struct ipkg_header {
+        char magic[4];
+        uint16 major;
+        uint16 minor;
+        uint32 toc_offset;
+        uint32 unknown_1;
+        uint32 toc_entries;
+        uint32 unknown_2[2];
+        uint32 always_null;
+        char file_version[256];
+        char product_name[256];
+        char ipkg_name[256];
+        char signature[256];
+    } ipkg_header_t;
+"""
+
+
+def is_valid_header(header: Instance) -> bool:
+    try:
+        snull(header.ipkg_name).decode("utf-8")
+        snull(header.file_version).decode("utf-8")
+        snull(header.product_name).decode("utf-8")
+    except UnicodeDecodeError:
+        return False
+    return True
+
+
+class HPIPKGExtractor(Extractor):
+    def __init__(self):
+        self._struct_parser = StructParser(C_DEFINITIONS)
+
+    def extract(self, inpath: Path, outdir: Path):
+        entries = []
+        with File.from_path(inpath) as file:
+            header = self._struct_parser.parse("ipkg_header_t", file, Endian.LITTLE)
+            file.seek(header.toc_offset, io.SEEK_SET)
+            for _ in range(header.toc_entries):
+                entry = self._struct_parser.parse(
+                    "ipkg_toc_entry_t", file, Endian.LITTLE
+                )
+                if not is_safe_path(outdir, Path(str(entry.name))):
+                    logger.warning(
+                        "Path traversal attempt, discarding.", output_path=entry.name
+                    )
+                    return
```

Comment:
Just skip this entry, not the entire thing.

```suggestion
                    continue
```